okta/user.go

```diff
@@ -346,18 +346,18 @@ func populateUserProfile(d *schema.ResourceData) *okta.UserProfile {
 }
 
 func setAdminRoles(d *schema.ResourceData, c *okta.Client) error {
+	roleTypes := make([]interface{}, 0)
+
 	// set all roles currently attached to user in state
 	roles, _, err := c.User.ListAssignedRoles(d.Id(), nil)
 
-	if err != nil {
-		return err
-	}
-
-	roleTypes := make([]interface{}, 0)
-	for _, role := range roles {
-		roleTypes = append(roleTypes, role.Type)
+	// only super admin can list roles if this fails assume the user has no
+	// admin roles
+	if err == nil {
```

Comment:
Love this idea. I think I would tweak this a bit. We should more specifically check the error to make sure it is indeed a permissions error. If it isn't, return the error. If it is, log a warning and not sync `admin_roles` in state.

I can merge this down to a branch that isn't forked, feel free to make that change if you get to it first. Thanks for the PR!

Reply:
thanks for the feedback. I've taken a second attempt at it.

Reply:
Great work!